builder/dockerfile/evaluator.go

```diff
@@ -34,17 +36,18 @@ import (
 )
 
 // Environment variable interpolation will happen on these statements only.
-var replaceEnvAllowed = map[string]bool{
-	command.Env:        true,
-	command.Label:      true,
-	command.Add:        true,
-	command.Copy:       true,
-	command.Workdir:    true,
-	command.Expose:     true,
-	command.Volume:     true,
-	command.User:       true,
-	command.StopSignal: true,
-	command.Arg:        true,
+var replaceEnvAllowed = map[reflect.Type]bool{
+	reflect.TypeOf((*command.EnvCommand)(nil)):        true,
```

Comment:
Hmm you mean in place expansion of typed commands fields ? That would also require to move all env variable expansion to the command package. Not sure if it is a lightweight change (and not sure if it is entirely feasible with build args logic)